### PR TITLE
Add vpcep endpoints data source

### PR DESCRIPTION
**Pull request by Benjile**

### CODE REVIEW

**Comment by niuzhenguo** [comment]
@Benjile Thanks for your contribution, please add documents of the new data source as well.

**Comment by Benjile** [comment]
I added the missing documentation
I also removed the error attribute because the underlying golandsk endpoint object does not seem to catch it: 
https://github.com/huaweicloud/golangsdk/blob/master/openstack/vpcep/v1/endpoints/results.go
If you have some snippets to do that i can implement it.

The documentation does not explain how to run the acceptance tests. Test requires a lot of environment variables.
Is is possible to have the execution context of those tests? I would like to run the tests by myself

File: docs/data-sources/vpcep_endpoints.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+---
+
+# flexibleengine\_vpcep\_endpoints
+
+Use this data source to get VPC endpoints.
+
+## Example Usage
+
+```hcl
+data "flexibleengine_vpcep_endpoints" "all_services" {
```
Comment (ShiChangkuo) [inline docs/data-sources/vpcep_endpoints.md:12]: maybe "all_endpoints" will be more clearly

File: docs/data-sources/vpcep_endpoints.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+data "flexibleengine_vpcep_endpoints" "dns_service" {
```
Comment (ShiChangkuo) [inline docs/data-sources/vpcep_endpoints.md:15]: maybe "dns_endpoints" will be more clearly

File: docs/data-sources/vpcep_endpoints.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  endpoint_service_name = "dns"
+}
+```
+
+## Argument Reference
+
+* `endpoint_service_name` - (Optional, String) Specifies the name of the VPC endpoint service.
```
Comment (ShiChangkuo) [inline docs/data-sources/vpcep_endpoints.md:22]: to keep compatible with `flexibleengine_vpcep_endpoint` resource, we can rename this argument to `service_name`.

File: docs/data-sources/vpcep_endpoints.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    The value is not case-sensitive and supports fuzzy match.
+
+* `endpoint_id` - (Optional, String) Specifies the unique ID of the VPC endpoint.
```
Comment (ShiChangkuo) [inline docs/data-sources/vpcep_endpoints.md:25]: as `flexibleengine_vpcep_endpoint` resource can be imported by id, does this argument is necessary?

File: docs/data-sources/vpcep_endpoints.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+* `id` - The unique ID of the public VPC endpoint service.
+* `status` - The connection status of the VPC endpoint.
+* `active_status` - The domain status.
+* `endpoint_service_name` - The name of the VPC endpoint service.
+* `endpoint_service_type` - The type of the VPC endpoint.
+* `endpoint_service_id` - The ID of the VPC endpoint service.
+* `vpc_id` - The ID of the VPC holding the VPC endpoint service.
+* `subnet_id` - The ID of the subnet holding the VPC endpoint.
+* `ip` - The IP of the VPC endpoint.
+* `marker_id` - The marker id of the VPC endpoint.
```
Comment (ShiChangkuo) [inline docs/data-sources/vpcep_endpoints.md:48]: I think some attributes are attributes empty or meaningless, how about refer to the attributes 
 of `flexibleengine_vpcep_endpoint` resource?

**Comment by ShiChangkuo** [comment]
## how to test

As you known, the acceptance test requires a lot of environment variables by [`testAccPreCheck`](https://github.com/FlexibleEngineCloud/terraform-provider-flexibleengine/blob/d32e040cbb6d4d0d5b3440d18ae5dc516d414ac0/flexibleengine/provider_test.go#L90).

you should also authentice you cloud account by user/password or AK/SK with environment variables, please refer to https://registry.terraform.io/providers/FlexibleEngineCloud/flexibleengine/latest/docs#configuration-reference.

then you can run the acceptance test by the following command:
```sh
make testacc TEST='./flexibleengine' TESTARGS='-run TestAccVPCEPEndpointsDataSourceBasic
```

File: docs/data-sources/vpcep_endpoints.md (hunk 4 above)
Comment (Benjile) [reply docs/data-sources/vpcep_endpoints.md:25]: I kept the same behaviour than other data sources of the provider. For example: 
https://registry.terraform.io/providers/FlexibleEngineCloud/flexibleengine/latest/docs/data-sources/vpcep_public_services

Getting informations about some objects via data sources and importing objects have not the same purpose;

Importing objects in a state file means we want to manage them. A data source is used when we do not need to manage the resource but when some of its attributes are needed

Comment (ShiChangkuo) [reply docs/data-sources/vpcep_endpoints.md:25]: @Benjile you are right, we can use `endpoint_id` to filter the endpoints in this data source.

File: docs/data-sources/vpcep_endpoints.md (hunk 1 above)
Comment (Benjile) [reply docs/data-sources/vpcep_endpoints.md]: done

File: docs/data-sources/vpcep_endpoints.md (hunk 2 above)
Comment (Benjile) [reply docs/data-sources/vpcep_endpoints.md]: done

File: docs/data-sources/vpcep_endpoints.md (hunk 3 above)
Comment (Benjile) [reply docs/data-sources/vpcep_endpoints.md]: done

File: docs/data-sources/vpcep_endpoints.md (hunk 5 above)
Comment (Benjile) [reply docs/data-sources/vpcep_endpoints.md]: done

**Comment by Benjile** [comment]
```console
make testacc TEST='./flexibleengine' TESTARGS='-run TestAccVPCEPEndpointsDataSourceBasic'
```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./flexibleengine -v -run TestAccVPCEPEndpointsDataSourceBasic -timeout 720m
=== RUN   TestAccVPCEPEndpointsDataSourceBasic
=== PAUSE TestAccVPCEPEndpointsDataSourceBasic
=== CONT  TestAccVPCEPEndpointsDataSourceBasic
--- PASS: TestAccVPCEPEndpointsDataSourceBasic (223.03s)
PASS
ok  	github.com/terraform-providers/terraform-provider-flexibleengine/flexibleengine	223.045s

**Comment by Benjile** [comment]
thanks!

**Pull request merged**